Authentication/SignUp/View/SignupController.swift

```diff
@@ -0,0 +1,230 @@
+            if let nameValidationMessageLabel = signupView.usernameValidationMessageLabel {
+                nameValidationMessageLabel.rex_text <~ _viewModel.nameValidationErrors.signal.map { $0.first ?? " " }
+            }
+        }
+    }
+    
+    func bindEmailElements() {
+        _viewModel.email <~ signupView.emailTextField.rex_textSignal
+        signupView.emailLabel?.text = _viewModel.emailText
+        signupView.emailTextField.placeholder = _viewModel.emailPlaceholderText
+        _viewModel.emailValidationErrors.signal.observeNext { [unowned self] errors in
+            self._delegate.signupController(self, didFailEmailValidationWithErrors: errors)
+        }
+        if let emailValidationMessageLabel = signupView.emailValidationMessageLabel {
+            emailValidationMessageLabel.rex_text <~ _viewModel.emailValidationErrors.signal.map { $0.first ?? " " }
+        }
+    }
+    
+    func bindPasswordElements() {
+        _viewModel.password <~ signupView.passwordTextField.rex_textSignal
+        signupView.passwordLabel?.text = _viewModel.passwordText
+        signupView.passwordTextField.placeholder = _viewModel.passwordPlaceholderText
+        _viewModel.passwordValidationErrors.signal.observeNext { [unowned self] errors in
+            self._delegate.signupController(self, didFailPasswordValidationWithErrors: errors)
+        }
+        if let passwordValidationMessageLabel = signupView.passwordValidationMessageLabel {
+            passwordValidationMessageLabel.rex_text <~ _viewModel.passwordValidationErrors.signal.map { $0.first ?? " " }
+        }
+        
+        _viewModel.passwordConfirmation <~ signupView.passwordConfirmTextField.rex_textSignal
+        signupView.passwordConfirmLabel?.text = _viewModel.confirmPasswordText
+        signupView.passwordConfirmTextField.placeholder = _viewModel.confirmPasswordPlaceholderText
+        _viewModel.passwordConfirmationValidationErrors.signal.observeNext { [unowned self] errors in
+            self._delegate.signupController(self, didFailPasswordConfirmationValidationWithErrors: errors)
+        }
+        if let passwordConfirmValidationMessageLabel = signupView.passwordConfirmValidationMessageLabel {
+            passwordConfirmValidationMessageLabel.rex_text <~ _viewModel.passwordConfirmationValidationErrors.signal.map { $0.first ?? " " }
+        }
+    }
+    
+    func bindButtons() {
+        signupView.signupButton.setTitle(_viewModel.signupButtonTitle, forState: .Normal)
+        signupView.signupButton.rex_pressed.value = _viewModel.signUpCocoaAction
+        signupView.signupButton.rex_enabled.signal.observeNext { [unowned self] in self.signupView.signupButtonEnabled = $0 }
+        
+        signupView.termsAndServicesLabel?.text = _viewModel.termsAndServicesLabelText
+        signupView.termsAndServicesButton.setTitle(_viewModel.termsAndServicesButtonTitle, forState: .Normal)
+        //signupView.termsAndServicesButton -> Presents the terms and services
```

Comment:
Because we haven't discussed yet how this would be. Maybe it's time to. @guidomb 
The terms and services should be a link and the button should take the user to the browser? Or something else?